### PR TITLE
[QNN] Concatenate operator

### PR DESCRIPTION
**Pull request by anijain2305**
@FrozenGene @u99127 @jackwish @tqchen 

Pinging for review. The code changes will be very small after we have requantize merged in.

### CODE REVIEW

**Comment by jackwish** [comment]
Thank you @anijain2305 , please remind me when this PR rebased to a merged requantize head :)

**Comment by anijain2305** [comment]
@u99127 @jackwish This is now ready for review.

**Comment by anijain2305** [comment]
@u99127 @tqchen Please review as you get time.

**Comment by anijain2305** [comment]
@u99127 @ZihengJiang @vinx13 Please review. This (hopefully) should be quick.

File: python/tvm/relay/qnn/op/qnn.py [hunk 1]
```diff
@@ -72,3 +73,75 @@ def requantize(data,
...
+    result: relay.Expr
+        The concatenated quantized tensor.
+    """
+
+    data = list(data)
+    requantized_exprs = list(data)
+
+    # Find the dtype of the input expr. This is required for the requantize op. Since, this is
+    # concatenate op, the dtype of the input is same as dtype of the output.
+    data0 = relay.transform.infer_type(data[0])
```
Comment (vinx13) [inline python/tvm/relay/qnn/op/qnn.py]: @anijain2305 This API has been removed in favor of `InferType`, please update and send another PR

**Comment by ZihengJiang** [comment]
It seems this PR has not passed the CI test? The new added test will raise error.

http://ci.tvm.ai:8080/blue/organizations/jenkins/tvm/detail/PR-3762/5/pipeline
http://ci.tvm.ai:8080/blue/organizations/jenkins/tvm/detail/master/1448/pipeline

**Comment by anijain2305** [comment]
@ZihengJiang @vinx13 Add the following quick fix for InferType 

https://github.com/dmlc/tvm/pull/3779

File: python/tvm/relay/qnn/op/qnn.py (hunk 1 above)
Comment (anijain2305) [reply python/tvm/relay/qnn/op/qnn.py]: Just added a PR for fixing InferType. I think it missed CI because CI passed 3 days ago and InferType changes happened after that.

Regarding `requantize`, it makes sense. Let me think little more tomorrow and finalize this.

Comment (zhiics) [reply python/tvm/relay/qnn/op/qnn.py]: I am trying to understand if we need to do type inference here. Shouldn't we have type for the input expr already? If not, shouldn't we infer type first?
